### PR TITLE
MINOR: set replication.factor to 1 to make StreamsBrokerCompatibility…

### PR DESCRIPTION
**Pull request by chia7712**
related to #10532

the default value of `replication.factor` was changed from `1` to `-1`. The old broker (< 2.4) does not support such configuration so this PR sets the `replication.factor` to `1` to fix `streams_broker_compatibility_test.py`. 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by ableegoldman — APPROVED** [review]
@mjsax or @chia7712 can you kick off this system test, or run it locally to verify the fix (if you haven't already)? Otherwise LGTM

File: tests/kafkatest/services/streams.py [hunk 1]
```diff
@@ -466,6 +466,15 @@ def __init__(self, test_context, kafka, processingMode):
                                                                 "org.apache.kafka.streams.tests.BrokerCompatibilityTest",
                                                                 processingMode)
 
+    def prop_file(self):
+        properties = {streams_property.STATE_DIR: self.PERSISTENT_ROOT,
+                      streams_property.KAFKA_SERVERS: self.kafka.bootstrap_servers(),
+                      # the old broker (< 2.4) does not support configuration replication.factor=-1
+                      "replication.factor": 1}
```
Comment (mjsax) [inline tests/kafkatest/services/streams.py:473]: Should we set it to 3 instead? IIRC, we run all system tests with 3 brokers? Wondering why the change broke the system tests, as they should have overwritten the default to 3 anyway?

Comment (ableegoldman) [reply tests/kafkatest/services/streams.py:473]: Sounds like we don't override the default after all? Or we have at least one test where that slipped through 🤷‍♀️

Comment (mjsax) [reply tests/kafkatest/services/streams.py:473]: I guess the logic scattered between Python and Java code... And maybe we need 3 only for EOS tests? Not sure either.

Also ok with me to just merge this PR as-is? Or should we update Java code instead?

Comment (ableegoldman) [reply tests/kafkatest/services/streams.py:473]: Ah, yeah, I bet that's it: we only set it for the EOS tests. Might be better to just set it in the Java code instead as it's easier to find and read, and I believe most other configs are set there. I think this test runs the `StreamsSmokeTest`?

Comment (chia7712) [reply tests/kafkatest/services/streams.py:473]: > Should we set it to 3 instead? IIRC, we run all system tests with 3 brokers?

not really. `streams_broker_compatibility_test.py` run test with single broker ( https://github.com/apache/kafka/blob/trunk/tests/kafkatest/tests/streams/streams_broker_compatibility_test.py#L45)

> Might be better to just set it in the Java code instead as it's easier to find and read, and I believe most other configs are set there. 

I prefer to change python code rather than java code since the number of brokers is connected to replication refactor. If we add hardcode (i.e `replication refactor = 1`) in the java class, it is hard to change both of them in python.

>  I think this test runs the StreamsSmokeTest?

BrokerCompatibilityTest (https://github.com/apache/kafka/blob/trunk/tests/kafkatest/services/streams.py#L466)

Comment (ableegoldman) [reply tests/kafkatest/services/streams.py:473]: Fine with me either way. Thanks for the fix

Comment (mjsax) [reply tests/kafkatest/services/streams.py:473]: Fine with me, too. Feel free to merge.

**Review by mjsax — APPROVED** [review]
